test/spec/services/angular-apimock.js

```diff
@@ -377,6 +376,45 @@ describe('Service: apiMock', function () {
 				});
 
 			});
+			
+			describe('enable query params', function () {
+
+				beforeEach(function() {
+					apiMockProvider.config({stripQueries: false});
+					// Set global flag to mock
+					$location.search('apiMock', true);
+				});
+
+				afterEach(function() {
+					apiMockProvider.config({stripQueries: true});
+				});
+
+				it('should NOT ignore query objects in request URL (path has /?)', function () {
+					defaultRequest.url = '/api/pokemon/?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request URL (path has only ?)', function () {
+					defaultRequest.url = '/api/pokemon?name=pikachu';
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should NOT ignore query objects in request OBJECT', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'name': 'pikachu', 'strength': 'electricity'};
+					defaultExpectPath = '/mock_data/pokemon/name=pikachu&strength=electricity.get.json';
+					expectHttpSuccess();
+				});
+
+				it('should sort query objects in alphabetical order', function () {
+					defaultRequest.url = '/api/pokemon';
+					defaultRequest.params = {'strength': 'electricity', 'name': 'pikachu'};
```

Comment:
Sadly the only interception points are the ones we use. It would've been nice if Angular normalized the http config object for us first at some point, but alas it doesn't.

ngResource abstracts $http, so as long as we can call $http in whatever way it supports then it should be fine.